channeld/channeld.c

```diff
@@ -202,6 +207,52 @@ static void billboard_update(const struct peer *peer)
 		       announce_status, shutdown_status);
 }
 
+/* used in billboard_update to say how many blocks to go! */
+static char *get_funding_depth(const struct peer *peer)
+{
+	status_trace("Please wait. We are trying to get depth...");
+	u8 *msg, *reply;
+	u32 minimum_depth, funding_depth;
+	char *infor = tal_arr(tmpctx, char, 0);
+	struct bitcoin_txid txid;
+
+	msg = towire_channel_funding_depth(tmpctx, &peer->channel->funding_txid);
+
+	/* wait for the reply, this loop may cost some minutes */
+	reply = master_wait_sync_reply(tmpctx,
+				       cast_const(struct peer *, peer),
+				       msg,
+				       WIRE_CHANNEL_FUNDING_DEPTH_REPLY);
+
+	if (!fromwire_channel_funding_depth_reply(reply,
+						  &txid,
+						  &minimum_depth,
+						  &funding_depth))
+		master_badmsg(WIRE_CHANNEL_FUNDING_DEPTH_REPLY, msg);
+
+	/* the struct channel in channeld.c and the struct channel in
+	** channel_control.c are different, there should compare txid
+	** to confirm channeld and master are saying the same
+	** (funding_txid)channel */
+	if(!bitcoin_txid_eq(&peer->channel->funding_txid, &txid))
```

Comment:
I see this is some sort of sanity check, but to me this seems unnecessary as our peer can have only one channel. If we can do without also makes msg simpler.

Reply:
I added this check considered the check in [`handle_peer_funding_locked()`](https://github.com/ElementsProject/lightning/blob/3e4a3bafc3db443c00a1bb16f95c0f0ca69850d4/channeld/channeld.c#L455). I think the check in  `handle_peer_funding_locked()` is similar with the check here. To be honest, I'm not sure...